operators/pkg/bastion-controller/bastion_controller.go

```diff
@@ -37,35 +38,41 @@ type BastionReconciler struct {
 	AuthorizedKeysPath string
 }
 
-func closeFile(f *os.File, log logr.Logger) {
-	err := f.Close()
-	if err != nil {
-		log.Error(err, "unable to close the file authorized_keys")
-	}
-}
-
 // +kubebuilder:rbac:groups=crownlabs.polito.it,resources=tenants,verbs=list
 
 func (r *BastionReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 	log := r.Log.WithValues("tenant", req.NamespacedName)
 	log.Info("reconciling bastion")
 
-	// Get tenants resources
-	var tenants crownlabsalpha1.TenantList
-	if err := r.List(ctx, &tenants); err != nil {
-		return ctrl.Result{}, client.IgnoreNotFound(err)
-	}
+	tenant := &crownlabsalpha1.Tenant{}
+	deleted := false
 
-	// Collect public keys of tenants
-	keys := make([]string, 0, len(tenants.Items))
-	for _, tenant := range tenants.Items {
-		keys = append(keys, tenant.Spec.PublicKeys...)
+	if err := r.Get(ctx, req.NamespacedName, tenant); err != nil {
+		deleted = true
```

Comment:
I would add a check that the error is actually NotExists, and not a different kind of error